tools/test/export/build_test.py

```diff
@@ -17,165 +17,257 @@
-                   created project files
+        sys.stdout.flush()
+        self.counter += 1
+        name_str = ('%s_%s_%s') % (test_case.mcu, test_case.ide, test_case.name)
+        self.display_counter("Exporting test case  %s::%s\t%s" % (test_case.mcu,
+                                                                  test_case.ide,
+                                                                  test_case.name))
 
-        Returns:
-            successes: a list of strings that contain the mcu, ide, test
-                       properties of a successful build test
-            skips: a list of strings that contain the mcu, ide, test properties
-                   of a skipped test (if the ide does not support mcu)
-            failures: a list of strings that contain the mcu, ide, test
-                       properties of a failed build test
+        try:
+            exporter = export(test_case.mcu, test_case.ide,
+                              project_id=test_case.id, zip_proj=None,
+                              clean=True, src=test_case.src,
+                              export_path=path.join(EXPORT_DIR,name_str),
+                              silent=True)
+            self.build_queue.put((exporter,test_case))
+        except TargetNotSupportedException:
+            self.skips.append("%s::%s\t%s" % (test_case.mcu, test_case.ide,
+                                              test_case.name))
+            # Check if the specified name is in all_os_tests
 
-        """
-        successes = []
-        failures = []
-        skips = []
-        for mcu, ide in self.mcu_ide_pairs:
-            for test in self.tests:
-                export_location, name, src, lib = setup_project(ide, mcu,
-                                                                program=test)
-                test_name = Test(test).id
-                try:
-                    exporter = export_project(src, export_location, mcu, ide,
-                                              clean=clean, name=name,
-                                              libraries_paths=lib)
-                    exporter.progen_build()
-                    successes.append("%s::%s\t%s" % (mcu, ide, test_name))
-                except FailedBuildException:
-                    failures.append("%s::%s\t%s" % (mcu, ide, test_name))
-                except TargetNotSupportedException:
-                    skips.append("%s::%s\t%s" % (mcu, ide, test_name))
-
-                ProgenBuildTest.handle_log_files(export_location, ide, name)
-                if clean:
-                    shutil.rmtree(export_location, ignore_errors=True)
-        return successes, failures, skips
 
+def check_valid_mbed_os(test):
+    """Check if the specified name is in all_os_tests
+    args:
+        test: string name to index all_os_tests
+    returns: tuple of test_name and source location of test,
+        as given by find_tests"""
+    all_os_tests = find_tests(ROOT, "K64F", "ARM")
```

Comment:
No, it will run the specified tests for all specified targets. This function unfortunately requires a target to return a dictionary of possible tests. However, most tests in the TESTS folder at the root of mbed-is should also support all other v5 boards (in addition to K64F). That is my understanding of that function thanks to @bridadan